### PR TITLE
Internal linkage kernels

### PR DESCRIPTION
**Pull request by robertmaynard**
## Description
Mark all cuspatial CUDA kernels with internal linkage to avoid runtime symbol collisions.

Also updated the CI tests to also verify that we add no new CUDA kernel with external linkage.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/include/cuspatial/detail/point_linestring_nearest_points.cuh [hunk 1]
```diff
@@ -54,7 +54,7 @@ template <class Vec2dItA,
           class OffsetIteratorB,
           class OffsetIteratorC,
           class OutputIt>
-void __global__
+void CUSPATIAL_KERNEL
```
Comment (harrism) [inline cpp/include/cuspatial/detail/point_linestring_nearest_points.cuh:57]: Let's swap these to match all other kernels?

**Review by harrism — APPROVED** [review]
One tiny suggestion, but looks good! Thanks.

File: cpp/include/cuspatial/detail/point_linestring_nearest_points.cuh (hunk 1 above)
Comment (robertmaynard) [reply cpp/include/cuspatial/detail/point_linestring_nearest_points.cuh]: Done

**Review by bdice — APPROVED** [review]
Approving CI changes.

**Pull request merged**